pkg/liquidity-mining/test/ChildChainGaugeRegistry.test.ts

```diff
@@ -0,0 +1,135 @@
+    });
+
+    registry = await deploy('ChildChainGaugeRegistry', {
+      args: [vault.address, pseudoMinter.address, gaugeFactory.address],
+    });
+
+    await vault.grantPermissionGlobally(await actionId(pseudoMinter, 'addGaugeFactory'), admin.address);
+    await vault.grantPermissionGlobally(await actionId(registry, 'addGauge'), admin.address);
+    await vault.grantPermissionGlobally(await actionId(registry, 'removeGauge'), admin.address);
+
+    await pseudoMinter.connect(admin).addGaugeFactory(gaugeFactory.address);
+
+    let tx = await gaugeFactory.create(mockBPT.address);
+    let event = expectEvent.inReceipt(await tx.wait(), 'GaugeCreated');
+    gauge = await deployedAt('ChildChainGauge', event.args.gauge);
+
+    tx = await otherGaugeFactory.create(mockBPT.address);
+    event = expectEvent.inReceipt(await tx.wait(), 'GaugeCreated');
+    otherGauge = await deployedAt('ChildChainGauge', event.args.gauge);
+  });
+
+  describe('addGauge', () => {
+    it('can add a valid gauge to the registry', async () => {
+      const tx = await registry.connect(admin).addGauge(gauge.address);
+
+      expectEvent.inReceipt(await tx.wait(), 'GaugeAdded', { gauge: gauge.address });
+      expect(await registry.totalGauges()).to.eq(1);
+      expect((await registry.getGauges(0, 1))[0]).to.eq(gauge.address);
+    });
+
+    it('only privileged account can add a gauge', async () => {
+      await expect(registry.connect(other).addGauge(gauge.address)).to.be.revertedWith('SENDER_NOT_ALLOWED');
+    });
+
+    it('cannot add the same gauge twice', async () => {
+      await registry.connect(admin).addGauge(gauge.address);
+      await expect(registry.connect(admin).addGauge(gauge.address)).to.be.revertedWith('GAUGE_ALREADY_REGISTERED');
+    });
+
+    it('reverts if gauge is not from the factory', async () => {
+      const maliciousGauge = await deploy('MockChildChainGauge', { args: ['test'] });
+      await maliciousGauge.setMockFactory(gaugeFactory.address);
+      await expect(registry.connect(admin).addGauge(maliciousGauge.address)).to.be.revertedWith(
+        'GAUGE_NOT_FROM_FACTORY'
+      );
+    });
+
+    it('reverts if gauge factory is invalid', async () => {
+      await expect(registry.connect(admin).addGauge(otherGauge.address)).to.be.revertedWith('INVALID_GAUGE_FACTORY');
+    });
+  });
+
+  describe('removeGauge', () => {
+    sharedBeforeEach('add gauge', async () => {
+      await registry.connect(admin).addGauge(gauge.address);
+    });
+
+    it('can remove a valid gauge from the registry', async () => {
+      const tx = await registry.connect(admin).removeGauge(gauge.address);
+
+      expectEvent.inReceipt(await tx.wait(), 'GaugeRemoved', { gauge: gauge.address });
+      expect(await registry.totalGauges()).to.eq(0);
+      await expect(registry.getGauges(0, 1)).to.be.revertedWith('END_INDEX_OUT_OF_BOUNDS');
+    });
+
+    it('reverts if gauge is not present', async () => {
```

Comment:
You can consider using `context` here as well, although it might be a bit of an overkill given that this case is a bit simpler than `add`.